### ISSUE TITLE
No way to get the length of a list

### ISSUE THREAD

**Issue by stevenrbrandt** [STEllAR-GROUP/phylanx#357]
This code works, but it prints a list with a 4 as the first element
````
block(
    define(a,[1,2,3,4]),
    cout(shape(a))
)
```
AFAIK, there's no way to get an item out of a list.
You might think that this would work, i.e. calling shape(a,0), but it gives an "out of range" error
```
block(
    define(a,[1,2,3,4]),
    cout(shape(a,0))
)
```

**Comment by hkaiser** [comment]
`shape()` can be used similarly to `numpy.shape` as it returns the dimensions of the array. It can't be used to re-shape an array, though. It is also not meant to return an element of a list. 

In short `shape(x)` returns: 
- `[]` for zero-dimensional arrays 
- `[len(x)]` for one-dimensional arrays and
- `[rows(x), columns(x)]` for matrices. 

Additionally, for matrices, `shape()` can be invoked with a second argument specifying the axis for which the dimension should be returned:
- `shape(matrix, 0)` -> `rows(matrix)` and 
- `shape(matrix, 1)` -> `columns(matrix)` 

Extracting an item out of a list should be possible using the `slice()` primitive.

**Comment by stevenrbrandt** [comment]
As stated above, these things do not work. Calling shape(a) gives me a list, but I can't extract element zero from it. shape(a,0) gives an error message.

**Comment by hkaiser** [comment]
> As stated above, these things do not work. Calling shape(a) gives me a list, but I can't extract element zero from it. shape(a,0) gives an error message.

You're right, the current behavior misses to support a second `axis` argument in the case the first argument is a list or one-dimensional array. I will add this functionality.

**Comment by hkaiser** [comment]
This has been fixed by merging #363
